README.md

```diff
@@ -20,16 +20,49 @@
+To deploy a new space, you will need to provide:
+- `voting_delay`: The delay between when a proposal is created, and when the voting starts. A `voting_delay` of 0 means that as soon as the proposal is created, anyone can vote. A `voting_delay` of 3600 means that voting will start `3600` seconds after the proposal is created.
+- `voting_duration`: The duration of the voting period, i.e. how long people will be able to vote for a proposal.
+- `proposal_threshold`: The amount of voting power needed to be able to create a new proposal. Used to avoid having anyone be able to create a proposal.
+- `voting_strategies`: A list of accepted voting strategy contracts. A voting strategy contract is used to determine the voting power of a user. More information in the [Voting Strategy](#Voting-Strategies) section.
+- `authenticators`: A list of accepted authenticators. Authenticators are used to authenticate users. For more information, refer to the [Authenticators](#Authenticators) section.
+
+Once a space has been created, users can create new proposals by calling the `propose` method (provided the caller has at least `proposal_threshold` voting power). Users don't directly interact with the `space` contract, but use one of the `authenticator` as a proxy. Once a proposal has been created, and the `voting_delay` has elapsed, users can then vote for the proposal (once again, using an `authenticator` as a proxy). Once the `voting_duration` has passed, votes are closed, and anyone can call `finalize_proposal` (this time, directly on the space contract!): this will finalize the proposal, count the votes for/against/abstain, and call the execution contract. More information about the execution in the [Execution Contract](#Execution-Contract) section.
+
+#### Voting Strategies
 
-Snapshot is open-sourced software licensed under the © [MIT license](LICENSE).
+Voting strategy contracts are the contracts used to determine the voting power of a user. The most common example is using the erc20 token balance of a user to determine his voting power. But we could imagine other voting strategies: owning a specific NFT, owning NFT of collection X and another NFT of collection Y, having participated in protocol xyz... the possibilities are endless! We provide the [single_slot_proof](contracts/starknet/strategies/single_slot_proof.cairo) which allows classic ERC20 and ERC721 balances on L1 (thanks to [fossil](https://github.com/OilerNetwork/fossil)) to be used as voting power, but feel free to create your own strategies! The interface of a strategy can be found [here](contracts/starknet/strategies/interface.cairo).
 
-Usage
------
+#### Authenticators
+
+Authenticators are the contracts in charge of authenticating users. This repository provides two useful authenticators:
+- [ethereum_authenticator](contracts/starknet/authenticator/ethereum.cairo) which will authenticate a user based on a message signed by Ethereum private keys.

```

Comment:
I would add "Sig" in the name to differentiate with the other authenticator that use Ethereum transaction from L1. Also I'm thinking the authenticator could be used for Polygon, Avalanche etc, we may want to call it differently than Ethereum, it could be based on the algo name instead like ECDSA what do you think?